pkgs/build-support/portable-service/default.nix

```
@@ -0,0 +1,118 @@
+{ pkgs, lib, stdenv }:
+/*
+  Create a systemd portable service image
+  https://systemd.io/PORTABLE_SERVICES/
+
+  Example:
+  pkgs.portableService {
+      name = "demo";
+      units = [ demo-service demo-socket ];
+  }
+*/
+{
+  # The name and version of the portable service. The resulting image will be
+  # created in result/$name_$version.raw
+  name
```

```suggestion
  pname
```
maybe just follow mkDerivation pname+version.